core/client/fs/src/main/java/alluxio/client/block/stream/GrpcDataReader.java

```diff
@@ -168,7 +169,19 @@ public void close() throws IOException {
         return;
       }
       mStream.close();
-      mStream.waitForComplete(mDataTimeoutMs);
+
+      // When a reader is closed, there is technically nothing the client requires from the server.
+      // However, the server does need to cleanup resources for a client close(), including closing
+      // or canceling any temp blocks. Therefore, we should wait for some amount of time for the
+      // server to finish cleanup, but it should not be very long (since the client is finished
+      // with the read). Also, if there is any error when waiting for the complete, it should be
+      // ignored since again, the client is completely finished with the read.
+      try {
+        // Wait a short time for the server to finish the close, and then let the client continue.
+        mStream.waitForComplete(5 * Constants.SECOND_MS);
```

Comment:
Some applications may depend on the closure to be done on the server side so they can perform other operations on the file safely. Having the reader skip the wait may cause subsequent operations to fail.

Reply:
If another client tries to read this block, there shouldn't be an issue, since they would only be grabbing a read lock, which should be allowed. I don't think another client can write lock the block, since Alluxio blocks are write-once-read-many. Can you think of any other problem behavior?

Ideally, I would have just removed this `waitForComplete` entirely, but there are some tests which require some async caching tasks to complete (even though async caching in general is best-effort). Because of these tests, waiting some short amount of time is beneficial for the tests.

Reply:
One possible issue is when the same client which just closes a stream tries to do another write operation such as rename or delete. If the stream close function does not wait until server release the lock, the subsequent write operation may fail when attempting to acquire a lock. This breaks the file stream contract that many applications assumes.